api/caching/tasks.py

```diff
@@ -104,11 +104,20 @@ def ban_url(instance):
 
 @app.task(max_retries=5, default_retry_delay=10)
 def update_storage_usage_cache(target_id):
-    AbstractNode = apps.get_model('osf.AbstractNode')
-
-    storage_usage_total = AbstractNode.objects.get(
-        guids___id=target_id,
-    ).files.aggregate(sum=models.Sum('versions__size'))['sum'] or 0
+    sql = """
+                SELECT sum(version.size) FROM osf_basefileversionsthrough AS obfnv
+                LEFT JOIN osf_basefilenode file ON obfnv.basefilenode_id = file.id
+                LEFT JOIN osf_fileversion version ON obfnv.fileversion_id = version.id
+                LEFT JOIN osf_guid guid ON file.target_object_id = guid.object_id
```

Comment:
I think you can skip this line if you make sure the target [here](https://github.com/CenterForOpenScience/osf.io/pull/9166/files#diff-a74ffc85f1b853e07331e93ebf2864e3L121) uses `.id` for the pk instead of the guid. You should only have to change one test as well.

Reply:
Nice catch! One less join.